### PR TITLE
feat(DIST-41): Add event data to onSubmit callback

### PR DESCRIPTION
**Pull request by LexSwed**
[DIST-41](https://jira.typeform.tf/browse/DIST-41)

- adds `takeSubmitEventData` which can be used to extend the details taken from the event. Currently it's just `response_id` passed from stakhanov (PR here: https://github.com/Typeform/stakhanov/pull/3481)
- adds tests for `takeSubmitData`
- makes `onSubmit` handlers to pass event to `takeSubmitEventData` to pass it to provided callback
- adds tests that `event` is passed to the util function `takeSubmitEventData`

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ module.exports = {
   },
   rules: {
     "comma-dangle": ["error", "never"],
-    complexity: 0
+    complexity: 0,
+    "react/jsx-handler-names": 0
```
Comment (antoniofull) [inline .eslintrc.js:14]: 👍

File: src/core/utils/index.js [hunk 2]
```diff
@@ -174,3 +174,9 @@ export const redirectToUrl = event => {
     document.body.removeChild(anchor)
   } catch (e) {}
 }
+
+export const takeSubmitEventData = (event) => {
+  const { landingId } = event.detail || {}
+
+  return landingId && { response_id: landingId }
```
Comment (mathio) [inline src/core/utils/index.js]: Why not make the `response_id` camel case (`responseId`)?

File: src/core/views/mobile-modal.spec.js [hunk 3]
```diff
@@ -67,6 +69,17 @@ describe('MobileModal', () => {
     expect(onSubmitSpy).not.toHaveBeenCalled()
   })
 
+  it('passes event data to onSubmit callback', () => {
+    const onSubmitSpy = jest.fn()
+    const takeSubmitEventDataSpy = jest.spyOn(utils, 'takeSubmitEventData')
+    const event = new CustomEvent('form-submit', { detail: { embedId: EMBED_ID } })
+
+    mount(<MobileModal embedId={EMBED_ID} onSubmit={onSubmitSpy} url={URL} />)
```
Comment (antoniofull) [inline src/core/views/mobile-modal.spec.js:77]: wouldn't shallow rendering be faster than mount?

File: src/core/utils/index.spec.js [hunk 4]
```diff
@@ -151,8 +151,7 @@ describe('Utilities', () => {
 
       expect(utils.takeSubmitEventData({ detail })).toEqual({ response_id: detail.landingId })
 
-      expect(utils.takeSubmitEventData({ detail: {} })).toEqual(undefined)
-      expect(utils.takeSubmitEventData({})).toEqual(undefined)
+      expect(utils.takeSubmitEventData({ detail: {} })).toEqual({ response_id: undefined })
```
Comment (antoniofull) [inline src/core/utils/index.spec.js:154]: I'd use camelCase for these kind of variables

File: src/core/utils/index.js (hunk 2 above)
Comment (LexSwed) [reply src/core/utils/index.js]: I was thinking to match [API](https://developer.typeform.com/responses/reference/retrieve-responses/) schema

**Comment by mathio** [comment]
Please update your commit message based on [guidelines](https://www.notion.so/typeform/Git-Guidelines-ab8a59603f104b928b515cd2f3355455#5c774fdce9da47468dc1ceb10388bc97)

tl;dr `If applied, this commit will <your title here>`

And squash your commits into 1.

File: src/core/utils/index.js (hunk 2 above)
Comment (LexSwed) [reply src/core/utils/index.js]: so that if you make api request, you get `response.response_id` which is equal to `event.response_id` - less cognitive overhead

Comment (mathio) [reply src/core/utils/index.js]: Makes sense 👍

File: src/core/utils/index.js [hunk 5]
```diff
@@ -174,3 +174,9 @@ export const redirectToUrl = event => {
     document.body.removeChild(anchor)
   } catch (e) {}
 }
+
+export const takeSubmitEventData = (event) => {
+  const { landingId } = event.detail || {}
+
+  return { response_id: landingId }
```
Comment (mathio) [inline src/core/utils/index.js:181]: Please comment here once you find out if [landingId is the same as responseId](https://typeform.slack.com/archives/CSZC3BMMH/p1588002748098400)

File: .eslintrc.js (hunk 1 above)
Comment (LexSwed) [reply .eslintrc.js:14]: okay, fully agree, dumb way of fighting the system

Comment (mathio) [reply .eslintrc.js]: I know we have a lint warning because of this. You can try to fix that if you want (I tried in past but had issues with this one) but its fine to leave it as is since its out of scope.

Comment (antoniofull) [reply .eslintrc.js]: You could also disable it just for the file that causes the warning

File: README.md [hunk 6]
```diff
@@ -117,6 +117,27 @@ typeformEmbed.makePopup(url, options)
   )
   ```
 
+### `onSubmit` event
+
+Callback function `onSubmit` receives `event` object with next properties:
+
+| Parameter       | Type   | Description          |
+| --------------- | ------ | -------------------- |
+| `submission_id` | string | ID of the submission |
```
Comment (mathio) [inline README.md:126]: This does not seem to match with the code. Its named `response_id` there.

Comment (LexSwed) [reply README.md:126]: good catch 👍

**Comment by mathio** [comment]
I have a proposal. 

1) We merge this. But before we do, let's call the returned ID a `landingId` since thats what this is. We can deal with other implications later. We could ask results team extend API to allow retrieving responses by `landingId`, but thats for later.

2) We dont merge this since `landingId` is kind of useless.

3) We dont merge this and create a spike ticket to actually return `responseId` here.

@LexSwed @Gotusso WDYT?

**Comment by LexSwed** [comment]
> I have a proposal.
> 
> 1. We merge this. But before we do, let's call the returned ID a `landingId` since thats what this is. We can deal with other implications later. We could ask results team extend API to allow retrieving responses by `landingId`, but thats for later.
> 2. We dont merge this since `landingId` is kind of useless.
> 3. We dont merge this and create a spike ticket to actually return `responseId` here.
> 
> @LexSwed @Gotusso WDYT?

I returned to my team, so would ask someone to take over the task.

I'm into 2, I don't know how many customers we're satisfying with it.
I'm into 1st, `landingId` is documented in response structures, so it's not something new for devs, so we could have an endpoint. and then we could deprecate it. However, if responses are added to DB asynchronously, we can have a case when someone requests response by `landingId` right after form submit, but submission is still in the queue and hence doesn't exists in DB. so need to ping BE devs: https://typeform.slack.com/archives/CSZC3BMMH/p1588065761099600

File: src/core/utils/index.js (hunk 5 above)
Comment (LexSwed) [reply src/core/utils/index.js:181]: we need centralised place for this :D 
but @pchiwan said they wanted to introduce `response_id` in `stakhanov`, it's in their backlog. so the issue will be solved as soon as they have it

Comment (mathio) [reply src/core/utils/index.js:181]: If this is in their backlog (and it is possible to retrieve `response_id` in stakhanov) I would wait for that.

**Comment by prilutskiy** [comment]
Embed script preview available: https://embed.typeform.com/f6809c62d25be9af3ebb2eaf556700dcea59fbd0/embed.js

**Comment by prilutskiy** [comment]
Embed script preview available: https://embed.typeform.com/de321c2a51ffa30af4c68ffa1511ddb39de5ac62/embed.js

**Comment by mathio** [comment]
I updated this PR and opened a [related PR in stakhanov](https://github.com/Typeform/stakhanov/pull/3481). Now actual `response_id` is passed, not `landing_id`.

@LexSwed and @Typeform/distribution please re-review.

File: src/core/make-fullscreen.js [hunk 7]
```diff
@@ -5,7 +5,8 @@ import {
   applyIOSIframeResizeHack,
   replaceExistingKeys,
   redirectToUrl,
-  noop
+  noop,
+  takeSubmitEventData
```
Comment (antoniofull) [inline src/core/make-fullscreen.js:9]: I would have called `getSubmitEventData` 😄

**Comment by prilutskiy** [comment]
Embed script preview available: https://embed.typeform.com/0643980aa29da76e2dcd4ce821916f49b3d895c0/embed.js

**Comment by travis-tf** [comment]
Embed script preview available: https://embed.typeform.com/a389e6f5a0ea642182d4e5272870d9914fb5ccbc/embed.js

**Comment by travis-tf** [comment]
Embed script preview available: https://embed.typeform.com/cf81b1083ef5d80cf9efd41f04080b720335dff7/embed.js
